Add blog for volume group snapshot v1beta2 (#51390)
kubernetes/website: content/en/blog/_posts/2025-08-27-volume-group-snapshot-beta-2.md

---
layout: blog
title: "Kubernetes v1.34: Moving Volume Group Snapshots to v1beta2"

> draft: true

Reviewer: ```suggestion
draft: true
date: 2025-XX-XX
```

just a nit

slug: kubernetes-v1-34-volume-group-snapshot-beta-2
author: >
   Xing Yang (VMware by Broadcom)
---

Volume group snapshots were [introduced](/blog/2023/05/08/kubernetes-1-27-volume-group-snapshot-alpha/)

> as an Alpha feature with the Kubernetes 1.27 release and moved to [Beta](https://kubernetes.io/blog/2024/12/18/kubernetes-1-32-volume-group-snapshot-beta/) in the Kubernetes 1.32 release.

Reviewer: ```suggestion
as an Alpha feature with the Kubernetes 1.27 release and moved to [Beta](/blog/2024/12/18/kubernetes-1-32-volume-group-snapshot-beta/) in the Kubernetes 1.32 release.
```

Author: Done

The recent release of Kubernetes v1.34 moved that support to a second beta.
The support for volume group snapshots relies on a set of
[extension APIs for group snapshots](https://kubernetes-csi.github.io/docs/group-snapshot-restore-feature.html#volume-group-snapshot-apis).
These APIs allow users to take crash consistent snapshots for a set of volumes.
Behind the scenes, Kubernetes uses a label selector to group multiple PersistentVolumeClaims
for snapshotting.
A key aim is to allow you restore that set of snapshots to new volumes and
recover your workload based on a crash consistent recovery point.

This new feature is only supported for [CSI](https://kubernetes-csi.github.io/docs/) volume drivers.

## What's new in Beta 2?

> While testing the beta version, we encountered an [issue](https://github.com/kubernetes-csi/external-snapshotter/issues/1271) where the restoreSize is not set for individual VolumeSnapshotContents and VolumeSnapshots if CSI driver does not implement ListSnapshots.

Author: Fixed

> We evaluated various options [here](https://docs.google.com/document/d/1LLBSHcnlLTaP6ZKjugtSGQHH2LGZPndyfnNqR1YvzS4/edit?tab=t.0) and decided to make this v1beta2 API change.

Reviewer: ```suggestion
We evaluated various options [here](https://docs.google.com/document/d/1LLBSHcnlLTaP6ZKjugtSGQHH2LGZPndyfnNqR1YvzS4/edit?tab=t.0) and decided to make this change releasing a new beta for the API.
```

Author: Done

> Specifically, a VolumeSnapshotInfo struct is added in v1beta2. VolumeSnapshotInfo contains information for an indivivual volume snapshot that is a member of a volume group snapshot.

Author: Done

> Specifically, a VolumeSnapshotInfo struct is added in v1beta2. VolumeSnapshotInfo contains information for an individual volume snapshot that is a member of a volume group snapshot.

Reviewer: ```suggestion
Specifically, a VolumeSnapshotInfo struct is added in v1beta2, it contains information for an individual volume snapshot that is a member of a volume group snapshot.
```

Author: Done

> A VolumeSnapshotInfoList, a list of VolumeSnapshotInfo, is added to VolumeGroupSnapshotContentStatus to replace VolumeSnapshotHandlePairList.

Author: Done

> `VolumeSnapshotInfoList`, a list of VolumeSnapshotInfo, is added to VolumeGroupSnapshotContentStatus, replacing VolumeSnapshotHandlePairList.

Reviewer: ```suggestion
VolumeSnapshotInfoList, a list of VolumeSnapshotInfo, is added to VolumeGroupSnapshotContentStatus, replacing VolumeSnapshotHandlePairList.
```

---
layout: blog

> title: "Kubernetes 1.34: Volume Group Snapshots Moves to v1beta2"

Author: Done

> date: 2025-08-27

Author: Done

> slug: kubernetes-1-34-volume-group-snapshot-beta-2

Author: Done
